webextension/privileged/ui/CloudDownloadsView.jsm

```diff
@@ -0,0 +1,738 @@
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+ChromeUtils.defineModuleGetter(this, "CloudStorage",
+  "resource://gre/modules/CloudStorage.jsm");
+ChromeUtils.defineModuleGetter(this, "Downloads",
+  "resource://gre/modules/Downloads.jsm");
+ChromeUtils.defineModuleGetter(this, "DownloadPaths",
+  "resource://gre/modules/DownloadPaths.jsm");
+ChromeUtils.defineModuleGetter(this, "FileUtils",
+  "resource://gre/modules/FileUtils.jsm");
+ChromeUtils.defineModuleGetter(this, "NetUtil",
+  "resource://gre/modules/NetUtil.jsm");
+ChromeUtils.defineModuleGetter(this, "OS",
+  "resource://gre/modules/osfile.jsm");
+ChromeUtils.defineModuleGetter(this, "PlacesUtils",
+  "resource://gre/modules/PlacesUtils.jsm");
+ChromeUtils.defineModuleGetter(this, "RecentWindow",
+  "resource:///modules/RecentWindow.jsm");
+
+const EXPORTED_SYMBOLS = ["CloudDownloadsView"];
+const CLOUD_SERVICES_PREF = "cloud.services.";
+const CLOUD_PROVIDER_DEFAULT_ICON = "default";
+
+var CloudDownloadsView = {
+  stylesURL: null,      // Property storing URL to clouddownloads css
+  providers: null,      // Property of type Map object storing providers data
+  isInitialized: false, // Property to track API has added required observers
+  notificationHTML: `
+    <hbox>
+      <image id='cloudDownloadTypeIcon'/>
+      <vbox id='cloudDownloadContainer'>
+        <description id='cloudDownloadTitle'/>
+        <hbox>
+          <description id='cloudDownloadDetail'/>
+          <label id='cloudDownloadPreference' class='text-link'/>
+        </hbox>
+        <radiogroup id='multiProviderSelect'/>
+      </vbox>
+    </hbox>
+    <hbox>
+      <button id='cloudDownloadCancel' />
+      <button id='cloudDownloadSave' default='true'/>
+    </hbox>`,
+
+  get browserWindow() {
+    return RecentWindow.getMostRecentBrowserWindow();
```

Comment:
@gijsk Firefox version 61 onwards RecentWindow module is replaced with BrowserWindowTracker. Would like your input if it's possible to do a version check here and use  BrowserWindowTracker.getTopWindow() for >61 version so that extension can work in nightly and beta. Thanks!

Reply:
I actually don't know. I mean, you can do a version check by checking `Services.appinfo.version` using `Services.vc.compare()` but it'd be better to check for the new module directly. I don't know how to do that, however. See https://mozilla.logbot.info/fx-team/20180503#c14705162 . Maybe @mikedeboer knows.